scipy/stats/tests/test_fit.py

```diff
@@ -140,3 +144,277 @@ def test_expon_fit():
+            stats.fit(self.dist, self.data, shape_bounds)
+
+        message = "The intersection of user-provided bounds for `n`"
+        with pytest.raises(ValueError, match=message):
+            stats.fit(self.dist, self.data)
+        shape_bounds = [(-np.inf, np.inf), (0, 1)]
+        with pytest.raises(ValueError, match=message):
+            stats.fit(self.dist, self.data, shape_bounds)
+
+    def test_loc_bounds_iv(self):
+        message = "`loc_bounds` must be a tuple containing exactly two..."
+        loc_bounds = (0, 1, 2)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(self.dist, self.data, self.shape_bounds_a,
+                      loc_bounds=loc_bounds)
+
+        message = "There are no values for `loc` on the interval..."
+        loc_bounds = (10, 0)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(self.dist, self.data, self.shape_bounds_a,
+                      loc_bounds=loc_bounds)
+
+        message = "There are no integer values for `loc` on the interval..."
+        loc_bounds = (0.4, 0.6)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(self.dist, self.data, self.shape_bounds_a,
+                      loc_bounds=loc_bounds)
+
+        message = "The intersection of user-provided bounds for `loc`"
+        loc_bounds = (-np.inf, np.inf)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(self.dist, self.data, self.shape_bounds_a,
+                      loc_bounds=loc_bounds)
+
+    def test_scale_bounds_iv(self):
+
+        message = "`binom` is an instance of `rv_discrete`..."
+        scale_bounds = (1, 10)
+        with pytest.warns(RuntimeWarning, match=message):
+            stats.fit(self.dist, self.data, self.shape_bounds_a,
+                      scale_bounds=scale_bounds)
+
+        message = "`scale_bounds` must be a tuple containing exactly two..."
+        scale_bounds = (0, 1, 2)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(stats.norm, self.data, scale_bounds=scale_bounds)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(stats.norm, self.data, [], scale_bounds=scale_bounds)
+
+        message = "There are no values for `scale` on the interval..."
+        scale_bounds = (10, 0)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(stats.norm, self.data, scale_bounds=scale_bounds)
+
+        message = "The intersection of user-provided bounds for `scale`"
+        scale_bounds = (1, np.inf)
+        with pytest.raises(ValueError, match=message):
+            stats.fit(stats.norm, self.data, scale_bounds=scale_bounds)
+
+    # these distributions are tested separately
+    skip_basic_fit = {'nhypergeom', 'boltzmann', 'nbinom',
+                      'randint', 'yulesimon', 'nchypergeom_fisher',
+                      'nchypergeom_wallenius'}
+    @pytest.mark.parametrize("dist_name", dict(distdiscrete))
```

Comment:
Could be 
```python3
@pytest.mark.parametrize("dist_name, shapes", distdiscrete)
```
but we can't really expect this test to pass for all distributions and all parameters with these arbitrary bounds. Indeed, some of the alternate parameters in the full `distdiscrete` lead `fit` to find different parameters (e.g. integer variables that are off by one or swapped due to a symmetry) and this test fails. These are hard to test for in a general way. Only testing with one set of parameters for each distribution seems sufficient to me.